### DISCUSSION TITLE
Your example (spread syntax for data from useData()) will not work in vike-vue

### DISCUSSION THREAD

**Discussion by Perlover** [vikejs/vike#1530] [Discussion]
https://github.com/vikejs/vike/blob/eb683534dd2d9d2419d696299a5fe8b87e614d90/docs/pages/useData/%2BPage.mdx#L21

I was agonizing for a long time trying to work with my code, which had exactly the same structure as yours, namely I was trying to do a "spread" for `data` inside a `<script setup>` for a component. There were errors. Until I figured out what the problem was (after spending hours, as I have little experience):

The vike-vue inside `useData` returns not an object, not a reactive proxy, etc.. It returns a **computed** property. Which means that the value of the computed data is stored in the `.value` property. **Only inside a template can you work with data properties** as object properties, which is what vue does behind the scene. And inside the component's code, your example will not work. Instead, you should do it like this: 

```
  // Inside a UI component
  const data = useData()
  const { name, price } = data.value
```

Or even may be:

```
  // Inside a UI component
  const data = useData()
  const { name, price } = isRefs(data.value)
```

I didn't suggest push request because I don't know what exactly is returned from other plugins from other freemworks. In fact, I think it's important to decide whether to return a computed property (`computed(() => data)`) or a reacitve object (`reacitve(data)`). This is a question for the developers of vike-vue, among others ( ping @AurelienLourot @brillout).

**Comment by brillout** [comment]
Thank you for sharing that. We actually reverted it: https://github.com/vikejs/vike-vue/commit/a5f189a984ff72422b818e468601ff990463c210. I stand to be corrected but I think we don't need `computed()`. The original motivation for using `computed()` was to make `data` reactive but I don't think we need this as `data` isn't expected to mutate.

**Reply by Perlover** [reply]
Apparently the change hasn't yet been reflected in the version on the npm repository, where the latest version is 9 days ago, and you reverted it in the last 7 days.

It is better, of course, that your code stays the one that is latest. In that case the examples will work. And I think the decision to make data reactive or not is better left to those who use the framework.

**Reply by brillout** [reply]
I just double checked and `vike-vue@0.6.3` does contain the commit (it doesn't use `computed()`).

**Reply by Perlover** [reply]
Yes, you are right!
My version (`pnpm list`) shows to me the v0.6.0
